### ISSUE TITLE
Race Condition?

### ISSUE THREAD

**Issue by ugjka** [thoj/go-ircevent#86]
Is this my own fault? (https://github.com/ugjka/reddit-irc-bot/blob/master/bot.go)

```[ugjka@ChakraPC oneshots]$ go run -race redditbot.go 
2016/11/05 17:40:51 Connected to irc.freenode.net:6666 (162.213.39.42:6666)
==================
WARNING: DATA RACE
Read at 0x00c42007c958 by goroutine 13:
  github.com/thoj/go-ircevent.(*Connection).pingLoop()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:167 +0x1f9

Previous write at 0x00c42007c958 by goroutine 11:
  github.com/thoj/go-ircevent.(*Connection).readLoop()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:77 +0x41f

Goroutine 13 (running) created at:
  github.com/thoj/go-ircevent.(*Connection).Connect()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:437 +0xd29
  github.com/ugjka/reddit-irc-bot.Start()
      /home/ugjka/goworks/src/github.com/ugjka/reddit-irc-bot/bot.go:156 +0x1a5
  main.main()
      /home/ugjka/goworks/src/oneshots/redditbot.go:23 +0xad

Goroutine 11 (running) created at:
  github.com/thoj/go-ircevent.(*Connection).Connect()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:435 +0xcdf
  github.com/ugjka/reddit-irc-bot.Start()
      /home/ugjka/goworks/src/github.com/ugjka/reddit-irc-bot/bot.go:156 +0x1a5
  main.main()
      /home/ugjka/goworks/src/oneshots/redditbot.go:23 +0xad
```

**Comment by ugjka** [comment]
ok this happens even with the most bare config
```go
package main

import (
  irc "github.com/thoj/go-ircevent"
)

const irc_nick = "_ugjka_"
const irc_name = "testing"
const irc_server = "irc.freenode.net:6667"
const irc_channel = "#ugjka"

func main(){
  ircobj := irc.IRC(irc_nick, irc_name)
	ircobj.Connect(irc_server)
	ircobj.Join(irc_channel)
  ircobj.Privmsg(irc_channel, "test test test test")
  ircobj.Loop()
}
```

Output:
```
[ugjka@ChakraPC oneshots]$ go run -race irctest.go 
2016/11/05 19:01:19 Connected to irc.freenode.net:6667 (94.125.182.252:6667)
==================
WARNING: DATA RACE
Read at 0x00c42007c698 by goroutine 13:
  github.com/thoj/go-ircevent.(*Connection).pingLoop()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:167 +0x1f9

Previous write at 0x00c42007c698 by goroutine 11:
  github.com/thoj/go-ircevent.(*Connection).readLoop()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:77 +0x41f

Goroutine 13 (running) created at:
  github.com/thoj/go-ircevent.(*Connection).Connect()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:437 +0xd29
  main.main()
      /home/ugjka/goworks/src/oneshots/irctest.go:14 +0x84

Goroutine 11 (running) created at:
  github.com/thoj/go-ircevent.(*Connection).Connect()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:435 +0xcdf
  main.main()
      /home/ugjka/goworks/src/oneshots/irctest.go:14 +0x84
==================
```

**Comment by thoj** [comment]
Probably from the library. There are some use of global flags in the code still that triggers this warning. I have yet to see an actual race condition though. Patches and PRs are welcome. 
On Nov 5, 2016 18:05, ugjka <notifications@github.com> wrote:ok this happens even with the most bare config

package main

import (
  irc "github.com/thoj/go-ircevent"
)

const irc_nick = "_ugjka_"
const irc_name = "testing"
const irc_server = "irc.freenode.net:6667"
const irc_channel = "#ugjka"

func main(){
  ircobj := irc.IRC(irc_nick, irc_name)
    ircobj.Connect(irc_server)
    ircobj.Join(irc_channel)
  ircobj.Privmsg(irc_channel, "test test test test")
  ircobj.Loop()
}

Output:

[ugjka@ChakraPC oneshots]$ go run -race irctest.go 
2016/11/05 19:01:19 Connected to irc.freenode.net:6667 (94.125.182.252:6667)
==================
WARNING: DATA RACE
Read at 0x00c42007c698 by goroutine 13:
  github.com/thoj/go-ircevent.(*Connection).pingLoop()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:167 +0x1f9

Previous write at 0x00c42007c698 by goroutine 11:
  github.com/thoj/go-ircevent.(*Connection).readLoop()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:77 +0x41f

Goroutine 13 (running) created at:
  github.com/thoj/go-ircevent.(*Connection).Connect()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:437 +0xd29
  main.main()
      /home/ugjka/goworks/src/oneshots/irctest.go:14 +0x84

Goroutine 11 (running) created at:
  github.com/thoj/go-ircevent.(*Connection).Connect()
      /home/ugjka/goworks/src/github.com/thoj/go-ircevent/irc.go:435 +0xcdf
  main.main()
      /home/ugjka/goworks/src/oneshots/irctest.go:14 +0x84
==================

**Comment by ugjka** [comment]
Ok thanks

**Comment by thoj** [comment]
Merged some fixes that where living in testing branch.

**Comment by ugjka** [comment]
@pdf reopening for you

**Comment by pdf** [comment]
Ah, sorry @ugjka looks like this was fixed by #97 but no release has been tagged in a long time.

**Comment by ugjka** [comment]
ok lol closing!
